### PR TITLE
Fix build warnings and update CI build checks to cover library header files

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR fixes warnings related to C standard compliance and updates the CMake files to include the +TCP library header files to be included in the build checks.

Updates the unit tests with respect to the changes made.

Test Steps
-----------
Fix incompatable code and build checks in CI.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_IP_Utils.c [hunk 1]
```diff
@@ -1764,3 +1767,18 @@ uint16_t usChar2u16( const uint8_t * pucPtr )
...
 /*-----------------------------------------------------------*/
+
+#if ( ipconfigUSE_DHCPv6 == 1 ) || ( ipconfigUSE_DHCP == 1 )
+
+/**
+ * @brief Returns the current state of a DHCP process.
+ *
+ * @param[in] pxEndPoint the end-point which is going through the DHCP process.
+ */
+    eDHCPState_t eGetDHCPState( const struct xNetworkEndPoint * pxEndPoint )
```
Comment (moninom1) [inline source/FreeRTOS_IP_Utils.c:1778]: Why is this needed to be moved out from DHCP.c?

Comment (tony-josi-aws) [reply source/FreeRTOS_IP_Utils.c:1778]: Its used by `xSendDHCPEvent` which is used by both DHCPv6 and DHCPv4

File: source/FreeRTOS_UDP_IPv6.c [hunk 2]
```diff
@@ -130,66 +130,26 @@ static eARPLookupResult_t prvStartLookup( NetworkBufferDescriptor_t * const pxNe
...
-            *pxLostBuffer = pdTRUE;
-        }
+        IPv6_Type_t eTargetType = xIPv6_GetIPType( &( pxNetworkBuffer->xIPAddress.xIP_IPv6 ) );
+        BaseType_t xIsGlobal = ( eTargetType == eIPv6_Global ) ? pdTRUE : pdFALSE;
+        pxNetworkBuffer->pxEndPoint = pxGetEndpoint( ( BaseType_t ) ipTYPE_IPv6, xIsGlobal );
+        FreeRTOS_printf( ( "prvStartLookup: Got an end-point: %s\n", pxNetworkBuffer->pxEndPoint ? "yes" : "no" ) );
     }
-    else
-    {
-        ulIPAddress = pxNetworkBuffer->xIPAddress.ulIP_IPv4;
```
Comment (moninom1) [inline source/FreeRTOS_UDP_IPv6.c:165]: Why are we removing IPv4 handling

Comment (tony-josi-aws) [reply source/FreeRTOS_UDP_IPv6.c:165]: It's duplicated dead code that is never executed. [This check](https://github.com/tony-josi-aws/FreeRTOS-Plus-TCP/blob/a497d48f1f5af632999bd52777d00707ec9a3b1c/source/FreeRTOS_UDP_IP.c#L108-L109) makes sure IPv4 handling is never required there.

File: source/FreeRTOS_IP_Utils.c (hunk 1 above)
Comment (moninom1) [reply source/FreeRTOS_IP_Utils.c:1778]: Ok, in future i think we should change DHCP.c to DHCPv4.c in any major release.

**Review by moninom1 — APPROVED** [review]

File: source/include/FreeRTOS_IP_Utils.h [hunk 3]
```diff
@@ -79,6 +79,13 @@ struct xNetworkInterface;
     BaseType_t xSendDHCPEvent( struct xNetworkEndPoint * pxEndPoint );
 #endif
 
+#if ( ipconfigUSE_DHCPv6 == 1 ) || ( ipconfigUSE_DHCP == 1 )
```
Comment (aggarg) [inline source/include/FreeRTOS_IP_Utils.h:82]:
```suggestion
#if ( ( ipconfigUSE_DHCPv6 == 1 ) || ( ipconfigUSE_DHCP == 1 ) )
```

File: test/unit-test/FreeRTOS_IP_Utils/FreeRTOS_IP_Utils_utest.c [hunk 4]
```diff
@@ -91,13 +91,14 @@ void test_xSendDHCPEvent( void )
...
 
-    eGetDHCPState_ExpectAnyArgsAndReturn( 12 );
+    xEndPoint.xDHCPData.eDHCPState = eInitialWait;
 
     xSendEventStructToIPTask_ExpectAnyArgsAndReturn( xResult );
 
     xReturn = xSendDHCPEvent( &xEndPoint );
 
     TEST_ASSERT_EQUAL( xResult, xReturn );
+    TEST_ASSERT_EQUAL( xEndPoint.xDHCPData.eExpectedState, eInitialWait );
```
Comment (aggarg) [inline test/unit-test/FreeRTOS_IP_Utils/FreeRTOS_IP_Utils_utest.c:101]: The order of the arugmnents needs to be swapped.

**Review by moninom1 — DISMISSED** [review]

File: source/FreeRTOS_UDP_IPv6.c (hunk 2 above)
Comment (moninom1) [reply source/FreeRTOS_UDP_IPv6.c:165]: Was this not caught in UT coverage test?

Comment (tony-josi-aws) [reply source/FreeRTOS_UDP_IPv6.c:165]: As the initial check was in a different file, no.

**Review by aggarg — APPROVED** [review]

**Review by moninom1 — APPROVED** [review]
